aten/src/ATen/native/cuda/ForeachUtils.cuh

```diff
@@ -0,0 +1,55 @@
+#pragma once
+#include <ATen/ATen.h>
+#include <ATen/native/cuda/Loops.cuh>
+#include <ATen/native/cuda/MemoryAccess.cuh>
+namespace at { 
+namespace native {
+namespace {
+
+static constexpr int64_t kILP = 4;
+static constexpr int64_t kChunkSize = 65536;
+static constexpr int64_t kBlockSize = 512;
+
+template<typename T>
+__device__ __forceinline__ bool is_aligned(T* p){
+  return ((uint64_t)p) % (kILP * sizeof(T)) == 0;
+}
+
+template<typename T>
+__device__ __forceinline__ void load_store(T* dst, T* src, int dst_offset, int src_offset){
+  using LT = at::native::memory::aligned_vector<T, kILP>;
+  ((LT*)dst)[dst_offset] = ((LT*)src)[src_offset];
+}
+
+}
+
+bool check_fast_route(TensorList tensors, Scalar scalar) {
+  TORCH_CHECK(tensors.size() > 0, "Tensor list must have at least one tensor.");
+  auto expected_dtype = tensors[0].dtype();
+  auto expected_strides = tensors[0].strides();
+
+  for (auto t : tensors) {
+    if (t.dtype() != expected_dtype) {
+      return false;
+    }
+
+    if (t.strides() != expected_strides) {
```

Comment:
The code here is not doing the same thing as I suggested in https://github.com/pytorch/pytorch/pull/41554#discussion_r461062467.

The code here compares every tensor in the list against the first in the list, which is needlessly strict (different tensors across this list don't interact with each other at all).  Their dtypes should match, and they should all be non_overlapping_and_and_dense, but their strides (and sizes) don't need to match.  Their strides and sizes need to match partners tensors in other lists (although for this particular op, there are no other lists).

Reply:
Done